pkg/server/plugin/keymanager/gcpkms/client_fake.go

```diff
@@ -324,11 +329,21 @@ func (k *fakeKMSClient) setIsKeyDisabled(ok bool) {
 	k.keyIsDisabled = ok
 }
 
-func (k *fakeKMSClient) setGetPublicKeyErr(fakeError error) {
+func (k *fakeKMSClient) setGetPublicKeySequentialErrs(fakeErrors ...error) {
```

Comment:
What do you think of having this function take in a length and a single error string and then generating the errors array here?

Reply:
So, we don't want the same error to be returned here. To reach the desired scenario that this PR proposes (retry mechanism), we need to return some errors and then return nil.

Reply:
I dont think you need nil in the array. Just return nil when you get to the end

Reply:
5641933